### PR TITLE
Feat create hypothesis on error

### PR DESCRIPTION
**Pull request by ElMehdiTailassan**

### CODE REVIEW

File: src/test/java/org/gridsuite/modification/server/BuildTest.java [hunk 1]
```diff
@@ -607,16 +606,62 @@ public void runBuildWithReportErrorTest() throws Exception {
...
+        );
+
+        // not built node we throw the exception on the postAction so we will not send log report
+        NetworkStoreListener listener6  = NetworkStoreListener.create(networkTest, TEST_NETWORK_ID, null, modificationRepository, equipmentInfosService, false, false);
+        assertThrows("unexpected error", RuntimeException.class, () ->
+            networkModificationService.doAction(listener6, () -> {return;}
+                , () -> {
+                    throw new RuntimeException("unexpected error");}
+                , NetworkModificationException.Type.MODIFICATION_ERROR, TEST_NETWORK_ID, reporter, subReporter)
+        );
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/BuildTest.java:666]: Not all cases are present

File: src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java [hunk 2]
```diff
@@ -1925,12 +1925,14 @@ public void testErrorRemovingDanglingSwitches() throws Exception {
...
-        mvcResult = mockMvc.perform(get("/v1/groups").contentType(MediaType.APPLICATION_JSON))
-            .andExpect(status().isOk()).andReturn();
+        mvcResult = mockMvc.perform(get("/v1/groups/{groupUuid}/modifications", TEST_GROUP_ID)).andExpectAll(
+                status().isOk(),
+                content().contentType(MediaType.APPLICATION_JSON))
+            .andReturn();
         resultAsString = mvcResult.getResponse().getContentAsString();
-        List<UUID> deletionsGroups = mapper.readValue(resultAsString, new TypeReference<>() { });
-        assertEquals(deletionsGroups, List.of());
+        List<ModificationInfos> modificationsTestGroupId = mapper.readValue(resultAsString, new TypeReference<>() { });
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/modification/server/ModificationControllerTest.java:1933]: Rename by  `modifications`

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 3]
```diff
@@ -830,33 +816,29 @@ public List<EquipmentModificationInfos> createLoadModification(UUID networkUuid,
...
-            }
 
-            // add the generator modification entity to the listener
-            listener.storeGeneratorModification(generatorModificationInfos);
-        }, MODIFY_GENERATOR_ERROR, repordId, reporter, subReporter).stream().map(EquipmentModificationInfos.class::cast)
+            try {
+                Generator generator = getGenerator(network, generatorModificationInfos.getEquipmentId());
+                // modify the generator in the network
+                modifyGenerator(generator, generatorModificationInfos, subReporter);
+            } catch (NetworkModificationException exc) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:833]: Remove it

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/modification/server/BuildTest.java [hunk 4]
```diff
@@ -607,16 +606,62 @@ public void runBuildWithReportErrorTest() throws Exception {
...
         Reporter subReporter = reporter.createSubReporter("AttributeModification", "Attribute modification");
+
+        //error during the build we catch no exception
+        assertEquals(List.of(),
+            networkModificationService.doAction(listener1, () -> {
+                throw new RuntimeException("unexpected error");
+            }, null, NetworkModificationException.Type.MODIFICATION_ERROR, TEST_NETWORK_ID, reporter, subReporter)
+        );
+
+        // incremental error: we throw the exception  on the Action  and we are in iterative mode, not builded node
```
[the remark this answers is not in the document]
Comment (ElMehdiTailassan) [reply src/test/java/org/gridsuite/modification/server/BuildTest.java:620]: it is on the action part

File: src/test/java/org/gridsuite/modification/server/BuildTest.java (hunk 1 above)
Comment (ElMehdiTailassan) [reply src/test/java/org/gridsuite/modification/server/BuildTest.java:666]: wich one are not implemented ?

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 3 above)
Comment (ElMehdiTailassan) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:833]: why ? I suggest that if it's for a refacto, we process it in another US

**Review by ElMehdiTailassan — COMMENTED** [review]
changes done

**Pull request closed without merging**